plugin/route53/alias_resolver.go

```diff
@@ -0,0 +1,39 @@
+package route53
+
+import (
+	"context"
+	"github.com/miekg/dns"
+)
+
+// AliasResolver helps us in resolving ALIAS records from AWS.
+type AliasResolver interface {
```

Comment:
Why is this needed? Why is it an interface? And why can't the normal "get records from AWS" do this?

Reply:
AWS doesn't provide the records, ALIASES are FQDNS.

This is all the information you would get from the AWS API regarding an alias. And AWS evaluates the IP addresses based on healthchecks so even if the DNSName in the record set below has 3 IP's if the health check fails for 1 of them, you will only get 2 when you query AWS dns server.

```
(*route53.ResourceRecordSet)(0xc000746180)({
  AliasTarget: {
    DNSName: "d21v7mkn0v2jz6.cloudfront.net.",
    EvaluateTargetHealth: false,
    HostedZoneId: "Z2FDTNDATAQYW2"
  },
  Name: "files.msp.sh.",
  Type: "A"
})
```

As for why it's an interface is so we can more easily test our code, without this it would actually go out and do queries when we run our tests.